### ISSUE TITLE
base_url adds trailing slash when there is no virtual directory, but does not when there is.

### ISSUE THREAD

**Issue by Jieiku** [Keats/gutenberg#1957]
# Bug Report
base_url adds trailing slash when there is no virtual directory, but does not when there is.

I have the following in my base.html:

```html
<base href="{{ config.base_url | safe }}">
```

If my config.toml is configured without a virtual directory then the trailing slash gets added.:

```toml
base_url = "https://abridge.netlify.app"
```

If I use a virtual directory, the trailing slash does NOT get added.:

```toml
base_url = "https://jieiku.github.io/abridge-stork"
```

The reason this is a problem is because It makes it difficult to dynamically load resources without coding in a workaround like checking for the presence of a trailing slash.

for instance here is a [snippet](https://github.com/Jieiku/abridge/blob/master/static/search_facade.js) where I dynamically loads some js:
```js
const baseUrl = document.baseURI;
loadSearch.src = baseUrl + 'search_bundle.min.js?h=' + sha256;
```

document.baseURI only works properly in this situation if the `<base>` tag is also set properly. The problem is that without a virtual directory I get a trailing slash, but with a virtual directory I do not.

I am not sure if maybe there is another reason for zola to do it this way, or if this is simply a bug.

## Environment

Zola version: 0.16.0

## Expected Behavior
I would think it would either always or never add a trailing slash to the baseUrl configured in config.toml

## Current Behavior
A trailing slash only gets added if there is not a virtual directory as part of the baseUrl

## Step to reproduce
You can zola build with a baseUrl defined with and without the virtual directory as part of the baseUrl and see this happening.

**Comment by Keats** [comment]
That's weird, I don't think we do anything with `base_url` at all?

**Comment by Jieiku** [comment]
hmm odd, now I am getting different results locally than I am on netlify.

Let me do a bit more testing, maybe netlify does something differently than what is happening locally.

**Comment by Jieiku** [comment]
When I reported this, zola serve base_url was different from the sites I had launched. At first I thought it was because of the virtual directory (I was launching a few demo sites at the time, some with and some without the virtual directory). After adding some debug statements into the javascript I have found that the real cause is that `const baseUrl = document.baseURI;` adds a trailing slash when a `:port` is part of the base_url (which is what happens when you use zola serve)

So this is a fault of the behavior of `document.baseURI` and not zola, I will do a bit more research, hopefully I can find a clean fix for this, otherwise I will have to write workaround into that javascript because I want the site to work the same regardless if deployed or running from zola serve.

For now this is the solution I am using:

```js
let baseUrl = document.baseURI;
if (baseUrl.slice(-1) == "/") {
    baseUrl = baseUrl.slice(0, -1);
}
```
